server/src/main/java/org/elasticsearch/plugins/UberModuleClassLoader.java

```diff
@@ -59,29 +57,21 @@ public class UberModuleClassLoader extends SecureClassLoader implements AutoClos
     private final ModuleLayer.Controller moduleController;
     private final Set<String> packageNames;
 
-    static UberModuleClassLoader getInstance(ClassLoader parent, String moduleName, List<Path> jarPaths) {
-        return getInstance(parent, moduleName, jarPaths, Set.of());
+    static UberModuleClassLoader getInstance(ClassLoader parent, String moduleName, Set<URL> jarUrls) {
```

Comment:
Is there a potential issue here with specific desired ordering of the JARs/URLs? By using a list, one can control which paths are considered first and which next, but unless a LinkedHashSet is used, the ordering would be random.

Reply:
That's a good question. Since this is intended for the plugin case, we've already been using the random ordering of the set when creating the plugin classloaders. It's possible that there's some other use case where we will want to control the order in which jars are scanned or processed when creating an uber-module (perhaps when we know that one jar is shadowed by another?). Can you think of a stronger reason to stick with the List? If so, I would defer to it.

Reply:
> Can you think of a stronger reason to stick with the List? If so, I would defer to it.

No, not really. If it's already random, then Set is better, we'll eliminate duplicates. I suppose in the future if we ever needed ordering we can supply a LinkedHashSet and the API will still work.

Reply:
Sounds good. Thanks!